### ISSUE TITLE
Feature request: way to get sync progress from API

### ISSUE THREAD

**Issue by stevedekorte** [Bitmessage/PyBitmessage#745]
It would be nice if there were a way to get some rough measure of sync progress via the API. Something like: 1) the number of total cached messages visible on connected nodes and 2) number of cached messages on local node. This way clients could poll it to display sync progress to the user. Otherwise, they think it's broken and see no reason to leave the app open while it syncs.

**Comment by PeterSurda** [comment]
This sounds interesting.

**Comment by stevedekorte** [comment]
Right now I just show # of messages processed, which at least let's users see that it's doing something. What I'd really like to show is: "number of messages other nodes have told me about but I don't have yet".

**Comment by PeterSurda** [comment]
Yes that's how I understood it too. I think it would be helpful to see the progress status and I also think it is doable.

**Comment by PeterSurda** [comment]
Do you want this shown in the UI, exposed to the API calls or both?

**Comment by stevedekorte** [comment]
In the API. We've written our own Bitmessage UI app called Bitpost: https://voluntary.net/bitpost/

We may be doing a JS port next year and moving to the JS implementation of Bitmessage.

**Comment by PeterSurda** [comment]
Look at the latest two commits in my fork (https://github.com/mailchuck/PyBitmessage/commit/53062f61b9eff29059bad81237161c8d4ee81ca1 and https://github.com/mailchuck/PyBitmessage/commit/f933cea697d47f700ab36c566fc6a6e31f36cfb3) for guidlines how to write the API change yourself. The commits expose the number of unsynced objects to the GUI in the network tab. I don't plan on making the API change myself before 0.6, but I can merge a commit if you want to do it.
